src/java.base/share/classes/java/lang/FdLibm.java

```diff
@@ -985,4 +985,213 @@ public static double compute(double x) {
+        private static final double ln2_lo      =  0x1.a39ef35793c76p-33; //  1.90821492927058770002e-10
+        private static final double invln2      =  0x1.71547652b82fep0;   //  1.44269504088896338700e+00
+        /* scaled coefficients related to expm1 */
+        private static final double Q1          = -0x1.11111111110f4p-5;  // -3.33333333333331316428e-02
+        private static final double Q2          =  0x1.a01a019fe5585p-10; //  1.58730158725481460165e-03
+        private static final double Q3          = -0x1.4ce199eaadbb7p-14; // -7.93650757867487942473e-05
+        private static final double Q4          =  0x1.0cfca86e65239p-18; //  4.00821782732936239552e-06
+        private static final double Q5          = -0x1.afdb76e09c32dp-23; // -2.01099218183624371326e-07
+
+        static double compute(double x) {
+            double y, hi, lo, c=0, t, e, hxs, hfx, r1;
+            int k, xsb;
+            /*unsigned*/ int hx;
+
+            hx  = __HI(x);  /* high word of x */
+            xsb = hx & 0x8000_0000;            /* sign bit of x */
+            y = Math.abs(x);
+            hx &= 0x7fff_ffff;               /* high word of |x| */
+
+            /* filter out huge and non-finite argument */
+            if (hx >= 0x4043_687A) {                  /* if |x| >= 56*ln2 */
+                if (hx >= 0x4086_2E42) {              /* if |x| >= 709.78... */
+                    if (hx >= 0x7ff_00000) {
+                        if (((hx & 0xf_ffff) | __LO(x)) != 0) {
+                            return x + x;     /* NaN */
+                        } else {
+                            return (xsb == 0)? x : -1.0; /* exp(+-inf)={inf,-1} */
+                        }
+                    }
+                    if (x > o_threshold) {
+                        return huge*huge; /* overflow */
+                    }
+                }
+                if (xsb != 0) { /* x < -56*ln2, return -1.0 with inexact */
+                    if (x + tiny < 0.0) {         /* raise inexact */
+                        return tiny - one;        /* return -1 */
+                    }
+                }
+            }
+
+            /* argument reduction */
+            if (hx > 0x3fd6_2e42) {           /* if  |x| > 0.5 ln2 */
+                if (hx < 0x3FF0_A2B2) {       /* and |x| < 1.5 ln2 */
+                    if (xsb == 0) {
+                        hi = x - ln2_hi;
+                        lo =  ln2_lo;
+                        k =  1;}
+                    else {
+                        hi = x + ln2_hi;
+                        lo = -ln2_lo;
+                        k = -1;
+                    }
+                } else {
+                    k  = (int)(invln2*x + ((xsb == 0) ? 0.5 : -0.5));
+                    t  = k;
+                    hi = x - t*ln2_hi;      /* t*ln2_hi is exact here */
+                    lo = t*ln2_lo;
+                }
+                x  = hi - lo;
```

Comment:
```suggestion
                x = hi - lo;
```

Reply:
> 

Thanks for the close reading; pushed those suggestions along with some others.